cloudfour/cloudfour.com-patterns
# Storybook Addon Viewport - Use Design Token Breakpoints

## Overview

Pulls in Theo breakpoints and assigns them as the viewports (with `height: 100%`) for `@storybook/addon-viewport`.

Retains the `defaultViewport` (Responsive, which is 100% x 100%) on page load and which is what the "Reset viewport" link returns to after a specific breakpoint viewport size is selected.

Note: The tokens have no "pretty" names, so the addon is defaulting to the names used elsewhere (e.g., `xs`, `m`, `l`).

If pretty names are desired, we should discuss where best to store / map those.

## Screenshots

![viewports](https://user-images.githubusercontent.com/771447/78606770-b46ce000-7812-11ea-89ed-1645633e7f91.gif)

## Testing

1. [Preview theme](https://deploy-preview-597--cloudfour-patterns.netlify.com/)
1. Verify that you can select among the viewports defined in Theo
1. Verify that the default is full-screen responsive and that you can return to that default by choosing "Reset viewport"

---

- Fixes [#580](https://github.com/cloudfour/cloudfour.com-patterns/issues/580)

/CC @tylersticka @dromo77
